tests/baselines/reference/declarationEmitInlinedDistributiveConditional.js

```diff
@@ -0,0 +1,57 @@
+//// [tests/cases/compiler/declarationEmitInlinedDistributiveConditional.ts] ////
+
+//// [test.ts]
+import {dropPrivateProps1, dropPrivateProps2} from './api';
+const a = dropPrivateProps1({foo: 42, _bar: 'secret'}); // type is {foo: number}
+//a._bar                                                // error: _bar does not exist           <===== as expected
+const b = dropPrivateProps2({foo: 42, _bar: 'secret'}); // type is {foo: number, _bar: string}
+//b._bar                                                // no error, type of b._bar is string   <===== NOT expected
+
+//// [api.ts]
+import {excludePrivateKeys1, excludePrivateKeys2} from './internal';
+export const dropPrivateProps1 = <Obj>(obj: Obj) => excludePrivateKeys1(obj);
+export const dropPrivateProps2 = <Obj>(obj: Obj) => excludePrivateKeys2(obj);
+
+//// [internal.ts]
+export declare function excludePrivateKeys1<Obj>(obj: Obj): {[K in PublicKeys1<keyof Obj>]: Obj[K]};
+export declare function excludePrivateKeys2<Obj>(obj: Obj): {[K in PublicKeys2<keyof Obj>]: Obj[K]};
+export type PublicKeys1<T> = T extends `_${string}` ? never : T;
+type PublicKeys2<T>        = T extends `_${string}` ? never : T;
+
+//// [internal.js]
+"use strict";
+exports.__esModule = true;
+//// [api.js]
+"use strict";
+exports.__esModule = true;
+exports.dropPrivateProps2 = exports.dropPrivateProps1 = void 0;
+var internal_1 = require("./internal");
+var dropPrivateProps1 = function (obj) { return (0, internal_1.excludePrivateKeys1)(obj); };
+exports.dropPrivateProps1 = dropPrivateProps1;
+var dropPrivateProps2 = function (obj) { return (0, internal_1.excludePrivateKeys2)(obj); };
+exports.dropPrivateProps2 = dropPrivateProps2;
+//// [test.js]
+"use strict";
+exports.__esModule = true;
+var api_1 = require("./api");
+var a = (0, api_1.dropPrivateProps1)({ foo: 42, _bar: 'secret' }); // type is {foo: number}
+//a._bar                                                // error: _bar does not exist           <===== as expected
+var b = (0, api_1.dropPrivateProps2)({ foo: 42, _bar: 'secret' }); // type is {foo: number, _bar: string}
+//b._bar                                                // no error, type of b._bar is string   <===== NOT expected
+
+
+//// [internal.d.ts]
+export declare function excludePrivateKeys1<Obj>(obj: Obj): {
+    [K in PublicKeys1<keyof Obj>]: Obj[K];
+};
+export declare function excludePrivateKeys2<Obj>(obj: Obj): {
+    [K in PublicKeys2<keyof Obj>]: Obj[K];
+};
+export declare type PublicKeys1<T> = T extends `_${string}` ? never : T;
+declare type PublicKeys2<T> = T extends `_${string}` ? never : T;
+export {};
+//// [api.d.ts]
+export declare const dropPrivateProps1: <Obj>(obj: Obj) => { [K in import("./internal").PublicKeys1<keyof Obj>]: Obj[K]; };
+export declare const dropPrivateProps2: <Obj>(obj: Obj) => { [K in keyof Obj extends infer T ? T extends keyof Obj ? T extends `_${string}` ? never : T : never : never]: Obj[K]; };

```

Comment:
1. Is it not a problem that the middle conditional type (the inner of the two synthesized ones) is distributive itself? It seems like it comes out the same, but in a way the semantics are slightly different here, right?—the ``T extends `_${string}` `` bit will never actually have anything to distribute over because the distribution is done one level up.
2. Whether or not (1) actually matters, you could use Ron’s new infer constraints here instead, which would reduce a level of nesting. That seems preferable to me, unless we’re specifically trying to avoid emitting that for back compat reasons, but... I think that’s a job for downlevel-dts.

Reply:
1. Extra layers of distributivity don't matter, so long as the right type variable is distributed over before the conditional logic occurs.
2. Yeah, I mention that in a comment in the code - we could, but we'd start producing declaration files incompatible with older versions of TS and I don't know if we wanna do that just yet. It's always something we could change down the line.

Reply:
The other thing I was curious about is whether we could ever just copy the unreachable type alias declaration to a local one. If we’re already copying its contents, why not just drop the whole declaration nearby with a unique name?

Reply:
Wherever we can, we do (specifically if we can preserve it as-is, in it's existing location). We only do inlining like this when the type comes from somewhere unreachable or elided (like inside a different module's locals or an elided function body). We don't copy from those scopes into the current scope because then we'd have to do 1. pull in its whole dependency tree of types (which may, in turn, depend on things we just can't copy, like signature type parameters), and 2. ensure everything is renamed in such a way where there's no name conflicts. It's just not something we do much of right now.

Reply:
> pull in its whole dependency tree of types

This is what I thought at first, but don’t we end up having to do that if we inline it too? Like the only part of the type that we _don’t_ copy is the declaration.